File: Anul 2/Semestrul 2/Metode de proiectare si programare/labs/Tema 2-proiect Csharp/proiect/proiect/exceptions/RepoException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace proiect.exceptions
{
    class RepoException:Exception
    {
        public RepoException() { }
        public RepoException(string message) : base(message) { }
        public RepoException(string message,Exception cause) : base(message,cause) { }

        public string getMessage()
        {
            return base.Message;
        }
    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 1-proiect Java/proiect/src/main/java/com/example/proiect/exceptions/RepoException.java
package com.example.proiect.exceptions;

public class RepoException extends Exception {

    public RepoException() {
        super();
    }

    public RepoException(String message) {
        super(message);
    }

    public RepoException(String message, Throwable cause) {
        super(message, cause);
    }

    public RepoException(Throwable cause) {
        super(cause);
    }

    protected RepoException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }

    @Override
    public String getMessage() {
        return super.getMessage();
    }
}
